### PR TITLE
Add display names using `@display` annotation and add user defined errors

### PR DESCRIPTION
**Pull request by sachinira**
Enhance error handling in Azure Cosmos DB connector.
Fix https://github.com/wso2-enterprise/choreo/issues/2954

   Add display name for listener
   Add display names for remote operations
   Add display name for parameters in the remote operations
   Add display name for the return value of the remote operations
   Provide additional information about errors.

Provide additional information about errors using user defined errors

## Automation tests
 - Integration tests
   > Done for all operations implemented

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

Ballerina SL Alpha 2
JDK 11

### CODE REVIEW

File: client_endpoint.bal [hunk 1]
```diff
@@ -19,17 +19,20 @@ import ballerina/http;
 # Azure Cosmos DB Client Object for executing data plane operations.
 # 
 # + httpClient - the HTTP Client
+@display {
+    label: "Azure Cosmos DB Client"
+}
```
Comment (LakshanSS) [inline client_endpoint.bal:24]: Shall we change it like this?
```
@display {label: "Azure Cosmos DB Client"}

File: client_endpoint.bal [hunk 2]
```diff
@@ -64,18 +73,23 @@ public client class DataPlaneClient {
...
-                                    int|float|decimal|string partitionKey, 
-                                    DocumentReplaceOptions? documentReplaceOptions = ()) 
-                                    returns @tainted Document|error {
+    # + documentReplaceOptions - The `DocumentReplaceOptions` which can be used to add additional capabilities to the 
+    #                            request
+    # + return - If successful, returns a `Document`. Else returns `Error`.
+    @display{
+        label: "Replace document"
+    } 
+    remote function replaceDocument(@display{label: "Database id"} string databaseId,
```
Comment (LakshanSS) [inline client_endpoint.bal:82]: Shall we add a space after the display kewword?
```
@display {label: "Database id"} string databaseId
```

File: client_endpoint.bal (hunk 1 above)
Comment (sachinira) [reply client_endpoint.bal]: Fixed. Thanks for the information

File: client_endpoint.bal (hunk 2 above)
Comment (sachinira) [reply client_endpoint.bal]: Fixed. Thanks for the information

**Comment by sachinira** [comment]
Closing as this is an Outdated PR. All changes are included in  https://github.com/ballerina-platform/module-ballerinax-azure-cosmosdb/pull/4

**Pull request closed without merging**